share/functions/__fish_config_interactive.fish

```diff
@@ -258,7 +258,8 @@ function __fish_config_interactive -d "Initializations that should be performed
     # Notify terminals when $PWD changes (issue #906).
     # VTE based terminals, Terminal.app, iTerm.app (TODO), and foot support this.
     if not set -q FISH_UNIT_TESTS_RUNNING
-        and test 0"$VTE_VERSION" -ge 3405 -o "$TERM_PROGRAM" = Apple_Terminal -a (string match -r '\d+' 0"$TERM_PROGRAM_VERSION") -ge 309 -o "$TERM_PROGRAM" = WezTerm -o "$TERM" = foot
+        and string match -q -- 'foot*' $TERM
+        or test 0"$VTE_VERSION" -ge 3405 -o "$TERM_PROGRAM" = Apple_Terminal -a (string match -r '\d+' 0"$TERM_PROGRAM_VERSION") -ge 309 -o "$TERM_PROGRAM" = WezTerm
```

Comment:
This groups wrong - if the `not set -q` is false, the `and` will be skipped, and the `or` could still be true. (you'd have the same issue in sh with `&&` and `||`)

Tbh I'd add an inner `if`:

```fish
if not set -q FISH_UNIT_TESTS_RUNNING
    if string match...
       or test
```

Reply:
(note: This is extra credit. If you don't want to do it, I'll merge the original version and add this myself)

Reply:
Gotcha. Should be fixed now (anything for extra credits 🤣 - though I'm not really sure I can take credit for this...)